cranelift/entity/src/set.rs

```diff
@@ -71,11 +74,11 @@ where
     /// `clear`ed or created with `new`.
     pub fn cardinality(&self) -> usize {
         let mut n: usize = 0;
-        for byte_ix in 0..self.len / 8 {
+        for byte_ix in 0..self.len / BITS {
             n += self.elems[byte_ix].count_ones() as usize;
         }
-        for bit_ix in (self.len / 8) * 8..self.len {
-            if (self.elems[bit_ix / 8] & (1 << (bit_ix % 8))) != 0 {
+        for bit_ix in (self.len / BITS) * BITS..self.len {
+            if (self.elems[bit_ix / BITS] & (1 << (bit_ix % BITS))) != 0 {
                 n += 1;
             }
         }
```

Comment:
While reviewing this patch I noticed that I'm not sure why the `cardinality` function was written the way it was. This is nothing wrong with your changes, but if you'd like to make another PR I think this function can be cleaned up.

As far as I can tell, `self.len` is supposed to always track the position of the last 1 bit in the set. If that's working correctly, then the first `count_ones` loop should work fine in all cases and we don't need a special case for the last element. The only trick is that instead of looping to `self.len/BITS`, we need to round up to the next element boundary with `(self.len + (BITS - 1)) / BITS`, like we've done elsewhere.

I don't want to make that change in this PR because it's possible that `self.len` is not being managed correctly and I'd want to think about that separately.